router-bridge/src/planner.rs

```diff
@@ -2466,4 +2466,26 @@ feature https://specs.apollo.dev/unsupported-feature/v0.1 is for: SECURITY but i
         )
         .unwrap());
     }
+
+    #[tokio::test]
+    async fn extracts_cost_directives() {
+        let schema = include_str!("testdata/custom_cost_schema.graphql");
+        let planner = Planner::<serde_json::Value>::new(schema.to_string(), Default::default())
+            .await
+            .expect("can create planner");
+        let subgraphs = planner.subgraphs().await.expect("can extract subgraphs");
+
+        let mut snapshot = String::new();
+        for name in vec!["subgraphWithCost", "subgraphWithListSize"] {
+            use std::fmt::Write;
+
+            let sdl = subgraphs.get(name).unwrap();
+            _ = writeln!(
+                &mut snapshot,
+                "--------------------\n{}\n--------------------\n{}",
+                name, sdl
+            );
+        }
+        insta::assert_snapshot!(snapshot);
```

Comment:
This is more a nit about consistency, but do we want to do something similar to [the other tests around `subgraphs()`](https://github.com/apollographql/federation-rs/pull/371)? E.g.
```rust
        let subgraphs = planner.subgraphs().await.unwrap();
        let subgraphs: BTreeMap<String, String> = subgraphs.into_iter().collect();
        for (name, schema) in subgraphs {
            insta::assert_snapshot!(schema);
        }
```

Reply:
I've been following the pattern used here https://github.com/apollographql/router/blob/dev/apollo-federation/tests/extract_subgraphs.rs#L93

Switching to `BTreeMap` should resolve the ordering issue I was solving by hard-coding the names through